TalZivelin/TZivelin-HW-W01.js

```diff
@@ -0,0 +1,125 @@
+1) (Difficulty: mixed) Predict the output of each of the following expressions, and explain the rules which determine it. If the output depends on the value of x, identify the conditions when the expression will output true. Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky! Don't trust your first instinct.
+
+a) "1" == 1 - True because this does not check for the var types.
+
+b) "1" === 1 - Flase, the === checks that both type (string, NUmber) and value are identical.
+
+c) x == 'x' - If X is a number this will be true.
+
+d) x == (x+'') - True, Adding an empty string to a variable will convert it to a string which is true.
+
+e) '' == ' ' -  True, a string with one space is equal to a string with one space
+
+f) x = true - True, x is assigned true, so x is true.
+
+g) var x; x == 'undefined' - False, undefined and 'undefined' are not the same thing.
+
+h) '9'<'10' - False, Javascript reads the first number for comparison so 9 is not smaller than 1.
+
+i) typeof x + 1 === "number" - True if x is a number false if x is a string.
+
+j) typeof x % 2 === "number" - True is x is a number that results in a number, otherwise false.
+
+k) typeof (x % 2) === "number" - True, you either get a number, or NaN which returns typeof "Number".
+
+l) x++ == ++x - False because by the time you get to ++x the original x is one number larger.
+
+m) ++x == x++ - True, because x was increased and then compared to an increase in x.
+
+n) "1"+x == 1+x - When there is a plus things are converted to string, if the second x is number this would be false, if it's a string this will be true.
+
+o) "0"+1 == 1 - True, 01 and 1 are the same.
+
+p) (typeof (x+1))===(typeof x) - True because if x is a number it will be a number type on both sides, if it;s a string it will be a string type both places.
+
+q) (x*1 == x) || ((typeof x) != "number") - If x is a number you will get true you will get true otherwise this is false.
+
+r) (x=(typeof (x+(typeof x))))==x - True after you are doing the typeof calculation you are assigning it to x so yes x will equal x.
+
+All of the following can be solved with ordinary expressions and global variables with primitive values. You don't need functions, loops, or other topics beyond our first two classes.
+
+2) (Difficulty: easy)
```

Comment:
2) These won't necesarily adjust the original values towards the mean: if the number is already greater than the mean (which will likely happen at least half the time), it will actually get farther from the mean. If the mean is 30, a value of 20 should become 25 and a value of 40 should become 35.